src/gt4py/analysis/passes.py

```diff
@@ -133,68 +110,44 @@ def make_k_intervals(transform_data: TransformData) -> List[IntervalInfo]:
-                        if axis_bound.offset < 0 or axis_bound.level == gt_ir.LevelMarker.END
-                        else 0
+                if offset < 0 and axis_bound.level != gt_ir.LevelMarker.END:
+                    raise IntervalSpecificationError(
+                        interval_def,
+                        "Invalid offset in interval specification",
+                        loc=axis_bound.loc,
                     )
-                    offset = axis_bound.offset
-
-                    if offset < 0 and axis_bound.level != gt_ir.LevelMarker.END:
-                        raise IntervalSpecificationError(
-                            interval_def,
-                            "Invalid offset in interval specification",
-                            loc=axis_bound.loc,
-                        )
-
-                    elif offset > 0 and axis_bound.level != gt_ir.LevelMarker.START:
-                        raise IntervalSpecificationError(
-                            interval_def,
-                            "Invalid offset in interval specification",
-                            loc=axis_bound.loc,
-                        )
-
-                # Update min sizes
-                if not 0 <= index <= transform_data.nk_intervals:
+                elif offset > 0 and axis_bound.level != gt_ir.LevelMarker.START:
                     raise IntervalSpecificationError(
                         interval_def,
-                        "Invalid variable reference in interval specification",
+                        "Invalid offset in interval specification",
                         loc=axis_bound.loc,
                     )
 
                 bounds[i] = (index, offset)
-                if index < transform_data.nk_intervals:
-                    transform_data.min_k_interval_sizes[index] = max(
-                        transform_data.min_k_interval_sizes[index], offset
-                    )
+                if index == 0:
+                    min_k_interval_size = max(min_k_interval_size, offset)
 
             if bounds[0][0] == bounds[1][0] - 1:
                 index = bounds[0][0]
                 min_size = 1 + bounds[0][1] - bounds[1][1]
-                transform_data.min_k_interval_sizes[index] = max(
-                    transform_data.min_k_interval_sizes[index], min_size
-                )
+                min_k_interval_size = max(min_k_interval_size, min_size)
 
             # Create computation intervals
             interval_info = IntervalInfo(*bounds)
             computation_intervals.append(interval_info)
 
+        # note(tehrengruber): initially min_k_interval_sizes was meant to store the minimal size of the respective
+        #  intervals inbetween axis splitters (for some reason excluding the last interval relative to LevelMarker.END).
+        #  Since dynamic splitters were removed only one interval remains, but we adhere to the old / initial format
+        #  in TransformData here.
```

Comment:
If this would indeed be the way to add dynamic splitters again I would agree. However this code path is very old and should be removed altogether in case of a future cleanup (skipped here as it involved refactoring the StageMerger which was not worth it). As such I took a backward looking perspective for the person potentially doing such a refactoring and wandering about the strange format. If that's not convincing I'll change it.

Reply:
Ok, fair enough. It's not ideal but I agree it should be deprecated soon anyway, so it's ok.